apix-rest-v1/src/main/java/eu/xenit/apix/rest/exceptions/FileExistsException.java

```diff
@@ -0,0 +1,97 @@
+package eu.xenit.apix.rest.exceptions;
+
+import eu.xenit.apix.data.NodeRef;
+
+public class FileExistsException extends RuntimeException {
+
+    private NodeRef source;
+    private NodeRef parent;
+    private String name;
+
+    public FileExistsException() {
+    }
+
+    public FileExistsException(String message) {
+        super(message);
+    }
+
+    public FileExistsException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public FileExistsException(Throwable cause) {
+        super(cause);
+    }
+
+    public FileExistsException(String message, Throwable cause, boolean enableSuppression,
+            boolean writableStackTrace) {
+        super(message, cause, enableSuppression, writableStackTrace);
+    }
+
+    public FileExistsException(String message, NodeRef source, NodeRef parent, String name) {
```

Comment:
Thanks for the change, it's a lot of new constructors :smile: but the one I actually meant was a constructor that takes in an Alfresco FileExistsException. My understanding was that this exception exists only to wrap Alfresco exception, correct? So by adding that the existing repetition is removed:
```
public FileExistsException(org.alfresco.service.cmr.model.FileExistsException alfrescoException, NodeRef source) {
    this.source = source;
    this.parent = new NodeRef(alfrescoException.getParentNodeRef().toString()),
    this.name = alfrescoException.getName());
}
```
Anyway, this is a small remark and if I missed some good counterargument, let me know.

Reply:
Ah I see.
Adding the original exception is considered best practice for maintaining the stacktrace to the source of the exception.
The duplicated fields are a coincidence of the assumption that an apix client would not know the alfresco librrary perse, and thus wouldnt be able to access the original fields.
But given that assumption there is also a point to ending the stacktrace at the apix level -and thus not passing on the original exception.

Reply:
Ah and not passing the alfresco Exception explicitly maintains the decoupling between apix and alfresco. So indeed one last constructor with just the source, parent and name fields would do the above improvement too.

tbh I first had the fields on the apix exception public to avoid the wildgrow of constructors, but if we're going by the books, they should be there.

Reply:
Made an additional consideration, I first put the apix exception in the rest module, but I think it'd be more appropriate if the custom exceptions are stored in the interface module.

Reply:
> Ah and not passing the alfresco Exception explicitly maintains the decoupling between apix and alfresco. 

True but balance it with the fact that it will always be used like that leading to the code duplication I'm trying to avoid here.
Also: it's also called **FileExistsException** so the coupling to the Alfresco exception is clear :smile: 

> I think it'd be more appropriate if the custom exceptions are stored in the interface module.

Agreed.